internal/core/src/exec/expression/Expr.h

```diff
@@ -137,15 +137,8 @@ class SegmentExpr : public Expr {
         for (size_t i = current_data_chunk_; i < num_data_chunk_; i++) {
             auto data_pos =
                 (i == current_data_chunk_) ? current_data_chunk_pos_ : 0;
-            int64_t size = 0;
-            if (segment_->type() == SegmentType::Growing) {
-                size = (i == (num_data_chunk_ - 1) &&
-                        active_count_ % size_per_chunk_ != 0)
-                           ? active_count_ % size_per_chunk_ - data_pos
-                           : size_per_chunk_ - data_pos;
-            } else {
-                size = segment_->chunk_size(field_id_, i) - data_pos;
-            }
+            // if segment is chunked, type won't be growing
+            int64_t size = segment_->chunk_size(field_id_, i) - data_pos;
```

Comment:
Actually I think current logic is weird, growing segment is also "chunked" but in a different way from chunked sealed segment.
And if we decide to  delete non-chunked sealed segment related code in the near future, better to leave it as it is in case we forget to add it back